warehouse/oidc/forms/gitlab.py

```diff
@@ -20,7 +20,7 @@
 
 # https://docs.gitlab.com/ee/user/reserved_names.html#limitations-on-project-and-group-names
 _VALID_GITLAB_PROJECT = re.compile(r"^[a-zA-Z0-9][a-zA-Z0-9-_.]*$")
-_VALID_GITLAB_NAMESPACE = re.compile(r"^[a-zA-Z0-9][a-zA-Z0-9-_.]*$")
+_VALID_GITLAB_NAMESPACE = re.compile(r"^[a-zA-Z0-9][a-zA-Z0-9-_./]*$")
```

Comment:
This permits trailing slashes `foobar/` and multiple repeated slashes `foo//bar`, both of which I think should be invalid.

Reply:
That is also the case also for the other special symbols ([src](https://docs.gitlab.com/ee/user/reserved_names.html#limitations-on-usernames-project-and-group-names)):
> Usernames, project or group slugs:
> - Cannot start or end with a special character.
> - Must not contain consecutive special characters.

And there are also other limitations we don't check (like the project ending with `.git` or `.atom`. At the time, the reasoning was that we wanted some validation for the user input that discarded simple errors but didn't need to be exhaustive about rejecting invalid input (since we're not hitting the GitLab API to validate them).

Should we have more complex regexes and/or validation functions to account for all of the possible invalid inputs?

Reply:
Let's merge this as-is and create a new issue to make the regexes a little more accurate.

Reply:
issue created here: https://github.com/pypi/warehouse/issues/15852